cgi-bin/ljtestlib.pl

```diff
@@ -0,0 +1,83 @@
+package LJ;
+
+use strict;
+no warnings 'uninitialized';
+
+
+BEGIN {
+    use Test::MockObject;
+
+    my $mock = Test::MockObject->new();
+
+    sub fake_lang_ml {
+	my ( $code, $vars ) = @_;
+	my $aopts = $vars->{'aopts'};
+	if ( $code == "cleanhtml.error.markup.extra" ) {
+	    return "[<strong>Error:</strong> Irreparable invalid markup ('&lt;" . $aopts . "&gt;') in entry. Owner must fix manually. Raw contents below.]";
+	}
+	return "[[[" . $code . "]]]". $aopts;
+	if ( $code == "cleanhtml.error.markup" ) {
+	    return '<a ' . $aopts .'>Error: Irreparable invalid markup in entry. Raw contents behind the cut.</a>';
+	}
+    }
+
+    sub fake_get_proxy_url {
+      return "http://proxy.url";
+    }
+
+    $mock->fake_module('LJ::Lang' => (
+	    ml => \&fake_lang_ml 
+	));
+    $mock->fake_module('DW::Proxy' => (
+	    get_proxy_url => \&fake_get_proxy_url
+	));
+    $mock->fake_module('LJ::EmbedModule' => ());
+
+    # ugly hack to shutup dependent libraries which sometimes want to bring in
+    # ljlib.pl (via require, ick!).  so this lets them know if it's recursive.
+    # we REALLY need to move the rest of this crap to .pm files.
```

Comment:
Yeah :)

Reply:
i'm not even sure how this works to be honest :) i copied it wholesale from ljlib.